### PR TITLE
Dart 2 migration guide updates

### PR DESCRIPTION
**Pull request by chalin**
Supersedes #1401; the changes requested on that PR have been made here.

Contributes to #1366 

Staged at https://angulardart-org-dev2.firebaseapp.com/dart-2

cc @kevmoo

### CODE REVIEW

File: src/dart-2.md [hunk 1]
```diff
@@ -2,34 +2,99 @@
...
+- [Tour of Heroes, part 5][angular-examples/toh-5]
+
+### `pubspec.yaml` {#pubspec-yaml}
+
+- Add new `dev_dependencies`:
+  - `build_runner: {{site.data.pubspec.dev_dependencies.build_runner}}`
+  - `build_test: {{site.data.pubspec.dev_dependencies.build_test}}`, if you are running tests
+  - `build_web_compilers: {{site.data.pubspec.dev_dependencies.build_web_compilers}}`
+- Drop `dev_dependencies`:
+  - <del>`browser`</del>
```
Comment (kevmoo) [inline src/dart-2.md]: The strike-through rendering is a bit much...

File: src/dart-2.md [hunk 2]
```diff
@@ -2,34 +2,99 @@
...
+- Drop all `transformers`:
+  - <del>`angular`</del>
+  - <del>`dart_to_js_script_rewriter`</del>
+  - <del>`test/pub_serve`</del>
+
+For example, here is a diff of
+[angular-examples/quickstart/pubspec.yaml][]
+with these changes applied.
+
+### `web/index.html` {#web-index-html}
```
Comment (kevmoo) [inline src/dart-2.md]: This could be `example/sample1.html` too

Be a bit more generic?

File: src/dart-2.md (hunk 1 above)
Comment (chalin) [reply src/dart-2.md]: I'd like to keep some visual cue distinguishing deleted elements from updated or added elements. How about something like this:
> screen shot 2018-03-15 at 16 30 40

If not, I can drop the \<del> wrapper entirely.

File: src/dart-2.md (hunk 2 above)
Comment (chalin) [reply src/dart-2.md]: How about if I keep `web/index.hml` but add a comment to the effect that similar changes will be necessary for any HTML file that has a \<script> element with Dart file as a `src`?

**Comment by chalin** [comment]
@kevmoo - once the nits are resolved, will this be good to go?

File: src/dart-2.md (hunk 2 above)
Comment (kevmoo) [reply src/dart-2.md]: Sure...

File: src/dart-2.md (hunk 1 above)
Comment (kevmoo) [reply src/dart-2.md]: that's better...

**Comment by kevmoo** [comment]
> @kevmoo - once the nits are resolved, will this be good to go?

Yup!

**Comment by kevmoo** [comment]
...well, let's see what @kwalrath thinks

File: src/dart-2.md (hunk 2 above)
Comment (chalin) [reply src/dart-2.md]: I'm open to other suggestions. It's just that `web/index.html` is more likely to catch the eye than, say, `example/sample1.html` IMHO.

**Comment by kevmoo** [comment]
I was thinking something more general as a header "HTML script" tag or
something

Just be clear that it's not JUST for web/index.html

**Comment by chalin** [comment]
I refreshed the staging server.

**Pull request merged**